chain/epoch-manager/src/shard_tracker.rs

```diff
@@ -17,8 +20,11 @@ type BitMask = Vec<bool>;
 #[derive(Clone)]
 pub struct ShardTracker {
     tracked_shards_config: TrackedShardsConfig,
-    /// Stores shard tracking information by epoch, only useful if TrackedState == Accounts
-    tracking_shards_cache: Arc<SyncLruCache<EpochId, BitMask>>,
+    /// Stores shard tracking information by epoch, only useful if TrackedShardsConfig == Accounts
+    tracked_accounts_shards_cache: Arc<SyncLruCache<EpochId, BitMask>>,
+    /// Only relevant if `TrackedShardsConfig == Shards(tracked_shards)`.
+    /// Caches whether a shard is a descendant of any of the `tracked_shards`.
+    descendant_of_tracked_shard_cache: Arc<Mutex<HashMap<ShardId, bool>>>,
```

Comment:
nit: in one you have `shards_cache` and in other `shard_cache` -> please pick one and use it in both places.

Reply:
Can you also clarify what exactly is stored in each type and why cache is needed? 

For accounts something like: The accounts cache stores a bitmask of tracked shards per epoch id. This information is stored in a cache to avoid recomputing it. 

For descendants something like: The descendants cache stores for each shard id whether it is a descendant of any of the tracked shards. It is needed to handle resharding where the node should keep tracking children and further descendants of the configured shards. This information is stored in a cache to avoid recomputing it.